src/tactic/lint/misc.lean

```diff
@@ -319,3 +320,31 @@ meta def linter.check_univs : linter :=
 Note: if the linter flags an automatically generated declaration `xyz._proof_i`, it means that
 the universe problem is with `xyz` itself (even if the linter doesn't flag `xyz`)",
   is_fast := tt }
+
+/-!
+## Linter for syntactic tautologies
+-/
+
+/--
+Checks whether a lemma is a declaration that `∀ a b ... z,e₁ = e₂` where `e₁` and `e₂` are equal
+exprs, we call declarations of this form syntactic tautologies.
+Such lemmas are (mostly) useless and sometimes introduced unintentionally when proving basic facts
+with rfl when elaboration results in a different term than the user intended.
+-/
+meta def syn_taut (d : declaration) : tactic (option string) := do
+  let l := d.type.pi_codomain,
```

Comment:
Trivial style thing, but if you just inline this in line 336 you can avoid the nested `do`.